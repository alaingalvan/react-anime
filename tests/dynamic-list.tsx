import React, {useState} from 'react';
import renderer from 'react-test-renderer';
import Anime from '../src/react-anime';

function App(props) {
    let [ state, setState ] = useState({ myList: [ 'First Entry' ] });
    return (
        <div>
            <a
                style={{
                    textAlign: 'center',
                    display: 'block',
                    cursor: 'pointer',
                    userSelect: 'none'
                }}
                onClick={() => setState({ myList: [ ...state.myList, 'New Entry ' + state.myList.length ] })}
            >
                Add Entry
            </a>

            <Anime opacity={[ 0, 1 ]} translateY={'1em'} delay={(e, i) => i * 300}>
                <h1>Entries</h1>
                {state.myList.map((e, i) => (
                    <p key={i}>
                        <a
                            onClick={(_) => setState({ myList: state.myList.filter((__, j) => i !== j) })}
                            style={{ textDecoration: 'none' }}
                        >
                            ‚ùå
                        </a>{' '}
                        {e}
                    </p>
                ))}
            </Anime>
        </div>
    );
}

//snapshot test
it('dynamic list', () => {
    const tree = renderer.create(
      <App/>,
      {
        createNodeMock: _ => ({ nodeType: false })
      }
    );
  
    expect(tree.toJSON()).toMatchSnapshot();
  });