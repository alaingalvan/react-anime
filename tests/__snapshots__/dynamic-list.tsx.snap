// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dynamic list 1`] = `
<div>
  <a
    onClick={[Function]}
    style={
      Object {
        "cursor": "pointer",
        "display": "block",
        "textAlign": "center",
        "userSelect": "none",
      }
    }
  >
    Add Entry
  </a>
  <div>
    <h1>
      Entries
    </h1>
  </div>
  <div>
    <p>
      <a
        onClick={[Function]}
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        ‚ùå
      </a>
       
      First Entry
    </p>
  </div>
</div>
`;
